fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -446,6 +429,42 @@ protected Action buildActionChain() {
         }
     }
 
+    private <T extends Resource> com.ibm.fhir.persistence.jdbc.dto.Resource createResourceDTO(String logicalId, int newVersionNumber,
+            Instant lastUpdated, T updatedResource) throws IOException, FHIRGeneratorException {
+
+        Timestamp timestamp = FHIRUtilities.convertToTimestamp(lastUpdated.getValue());
+
+        com.ibm.fhir.persistence.jdbc.dto.Resource resourceDTO = new com.ibm.fhir.persistence.jdbc.dto.Resource();
+        resourceDTO.setLogicalId(logicalId);
+        resourceDTO.setVersionId(newVersionNumber);
+        resourceDTO.setLastUpdated(timestamp);
+        resourceDTO.setResourceType(updatedResource.getClass().getSimpleName());
+
+        // Most resources are well under 10K after being serialized and compressed
+        InputOutputByteStream ioStream = new InputOutputByteStream(DATA_BUFFER_INITIAL_SIZE);
+
+        // Serialize and compress the Resource
+        try (GZIPOutputStream zipStream = new GZIPOutputStream(ioStream.outputStream())) {
+            FHIRGenerator.generator(Format.JSON, false).generate(updatedResource, zipStream);
+            zipStream.finish();
+            resourceDTO.setDataStream(ioStream);
+        }
+
+        return resourceDTO;
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T extends Resource> T copyAndSetResourceMetaFields(T resource, int newVersionNumber, Instant lastUpdated) {
+        Meta meta = resource.getMeta();
+        Meta.Builder metaBuilder = meta == null ? Meta.builder() : meta.toBuilder();
+        metaBuilder.versionId(Id.of(Integer.toString(newVersionNumber)));
+        metaBuilder.lastUpdated(lastUpdated);
+
+        Builder resourceBuilder = resource.toBuilder();
+        resourceBuilder.setValidating(false);
```

Comment:
setting this to false will prevent unnecessary parse validation for a resource that we already know was valid